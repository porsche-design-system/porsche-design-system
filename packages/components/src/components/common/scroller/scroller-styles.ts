import { getCss } from '../../../utils';
import { addImportantToEachRule, getThemedColors, pxToRemWithUnit } from '../../../styles';
import type { ThemeExtendedElectric } from '../../../utils';
import type { GradientColorTheme } from './scroller-utils';
import type { JssStyle } from 'jss';

const actionButtonStyles: JssStyle = {
  display: 'flex',
  position: 'absolute',
  top: 0,
  height: '100%',
  alignItems: 'center',
};

export const getComponentCss = (
  gradientColorScheme: GradientColorTheme,
  theme: ThemeExtendedElectric,
  isNextHidden: boolean,
  isPrevHidden: boolean,
  prevNextButtonStyle?: JssStyle
): string => {
  const { backgroundColor, backgroundSurfaceColor } = getThemedColors(theme);
  const gradientColor = gradientColorScheme === 'surface' ? backgroundSurfaceColor : backgroundColor;
  const gradientColorTransparent = gradientColor + (gradientColor.length === 4 ? '0' : '00');

  return getCss({
    '@global': {
      ':host': addImportantToEachRule({
        display: 'block',
        height: '100%',
      }),
    },
    root: {
      position: 'relative',
      margin: `0 ${pxToRemWithUnit(-4)}`,
      height: 'inherit',
    },
    'scroll-area': {
      height: 'inherit',
      boxSizing: 'border-box',
      padding: pxToRemWithUnit(4),
      overflowY: 'hidden',
      overflowX: 'scroll',
      whiteSpace: 'nowrap',
      msOverflowStyle: 'none' /* IE and Edge */,
      scrollbarWidth: 'none' /* Firefox */,
      '&::-webkit-scrollbar': {
        display: 'none',
      },
    },
    // Extra wrapper needed to compensate different offset parent calculation depending of browser.
    // Needed for position of status bar.
    'scroll-wrapper': {
      position: 'relative',
      display: 'inline-block',
      minWidth: '100%',
    },
    trigger: {
      display: 'block',
      position: 'absolute',
      top: 0,
      height: '100%',
      width: pxToRemWithUnit(1),
      visibility: 'hidden',
      '&:first-of-type': {
        left: 0,
      },
      '&:last-of-type': {
        right: 0,
      },
    },
    'action-prev': {
      ...actionButtonStyles,
      left: 0,
      justifyContent: 'flex-start',
      '& $gradient': {
        background: `linear-gradient(90deg, ${gradientColor} 50%, ${gradientColorTransparent} 100%)`,
      },
      visibility: isPrevHidden ? 'hidden' : 'visible',
    },
    'action-next': {
      ...actionButtonStyles,
      right: 0,
      justifyContent: 'flex-end',
      '& $gradient': {
        background: `linear-gradient(90deg, ${gradientColorTransparent} 0%, ${gradientColor} 50%)`,
      },
      visibility: isNextHidden ? 'hidden' : 'visible',
    },
    button: {
      position: 'absolute',
      top: '50%',
      transform: 'translate3d(0,-50%,0)',
      ...prevNextButtonStyle,
    },
    gradient: {
      position: 'absolute',
      top: 0,
      bottom: 0,
      width: '2em',
      pointerEvents: 'none',
    },
  });
};
