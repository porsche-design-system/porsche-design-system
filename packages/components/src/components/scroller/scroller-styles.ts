import { getCss } from '../../utils';
import { addImportantToRule, getThemedColors, pxToRemWithUnit } from '../../styles';
import type { ThemeExtendedElectric } from '../../types';
import type { GradientColorTheme } from './scroller-utils';
import { getFocus } from '@porsche-design-system/utilities-v2';
import type { ScrollIndicatorPosition } from './scroller-utils';

export const getComponentCss = (
  gradientColorScheme: GradientColorTheme,
  isNextHidden: boolean,
  isPrevHidden: boolean,
  scrollIndicatorPosition: ScrollIndicatorPosition,
  theme: ThemeExtendedElectric
): string => {
  const { backgroundColor, backgroundSurfaceColor, baseColor } = getThemedColors(theme);
  const gradientColor = gradientColorScheme === 'surface' ? backgroundSurfaceColor : backgroundColor;
  const gradientColorTransparent = gradientColor + (gradientColor.length === 4 ? '0' : '00');

  const actionPrevNextStyles = {
    position: 'relative',
    padding: `${pxToRemWithUnit(4)} 0`,
    pointerEvents: 'none',
    display: 'flex',
    alignItems: scrollIndicatorPosition === 'center' ? 'center' : 'flex-start',
  };

  return getCss({
    '@global': {
      ':host': {
        display: 'block',
        height: addImportantToRule('inherit'),
      },
    },
    root: {
      display: 'grid',
      gridTemplateColumns: '2em minmax(0, 1fr) 2em',
      margin: `0 ${pxToRemWithUnit(-4)}`,
      height: 'inherit',
    },
    'scroll-area': {
      minHeight: pxToRemWithUnit(24),
      gridArea: '1 / 1 / 1 / -1',
      padding: pxToRemWithUnit(4),
      overflow: 'scroll hidden',
      msOverflowStyle: 'none' /* IE and Edge */,
      scrollbarWidth: 'none' /* Firefox */,
      '&::-webkit-scrollbar': {
        display: 'none',
      },
    },
    // Extra wrapper needed to compensate different offset parent calculation depending of browser.
    // Needed for position of status bar.
    'scroll-wrapper': {
      ...getFocus({ color: baseColor }),
      position: 'relative',
      display: 'inline-flex',
      minWidth: '100%',
      verticalAlign: 'top',
    },
    trigger: {
      position: 'absolute',
      top: 0,
      bottom: 0,
      width: '1px',
      visibility: 'hidden',
      '&:first-of-type': {
        left: 0,
      },
      '&:last-of-type': {
        right: 0,
      },
    },
    'action-prev': {
      ...actionPrevNextStyles,
      gridArea: '1 / 1 / 1 / 1',
      justifyContent: 'flex-start',
      background: `linear-gradient(90deg, ${gradientColor} 50%, ${gradientColorTransparent} 100%)`,
      visibility: isPrevHidden ? 'hidden' : 'visible',
      '& .button::before': {
        left: 0,
      },
    },
    'action-next': {
      ...actionPrevNextStyles,
      gridArea: '1 / 3 / 1 / 3',
      justifyContent: 'flex-end',
      background: `linear-gradient(90deg, ${gradientColorTransparent} 0%, ${gradientColor} 50%)`,
      visibility: isNextHidden ? 'hidden' : 'visible',
      '& .button::before': {
        right: 0,
      },
    },
    button: {
      pointerEvents: 'auto',
      position: 'static',
      // Pseudo-element to stretch the click-area to full height
      '&::before': {
        content: '""',
        position: 'absolute',
        top: 0,
        bottom: 0,
        width: 'max(2rem, 80%)',
      },
    },
  });
};
