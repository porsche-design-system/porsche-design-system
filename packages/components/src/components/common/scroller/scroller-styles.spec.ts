import { getComponentCss } from './scroller-styles';

describe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([
    ['default', false, false, 'center', 'light'],
    ['surface', false, false, 'center', 'dark'],
    ['default', false, false, 'center', 'light-electric'],
    ['surface', false, false, 'center', 'light'],
    ['default', false, false, 'center', 'dark'],
    ['surface', false, false, 'center', 'light-electric'],
    ['default', false, false, 'top', 'light'],
    ['default', true, false, undefined, 'light'],
    ['default', false, true, undefined, 'dark'],
    ['default', true, true, undefined, 'light'],
  ])(
    'should return correct css for gradientColorScheme: %s, isNextHidden: %s, isPrevHidden: %s and scrollIndicatorPosition: %s, theme: %s ',
    (...args) => {
      expect(getComponentCss(...args)).toMatchSnapshot();
    }
  );
});
