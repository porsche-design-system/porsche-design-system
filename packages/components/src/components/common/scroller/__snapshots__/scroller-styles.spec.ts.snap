// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for gradientColorScheme: default, isNextHidden: false, isPrevHidden: false and scrollIndicatorPosition: center, theme: dark  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #fff;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #0e1418 50%, #0e141800 100%);
  visibility: visible;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #0e141800 0%, #0e1418 50%);
  visibility: visible;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: default, isNextHidden: false, isPrevHidden: false and scrollIndicatorPosition: center, theme: light  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #000;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #fff 50%, #fff0 100%);
  visibility: visible;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #fff0 0%, #fff 50%);
  visibility: visible;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: default, isNextHidden: false, isPrevHidden: false and scrollIndicatorPosition: center, theme: light-electric  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #000;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #fff 50%, #fff0 100%);
  visibility: visible;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #fff0 0%, #fff 50%);
  visibility: visible;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: default, isNextHidden: false, isPrevHidden: false and scrollIndicatorPosition: top, theme: light  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #000;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: flex-start;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #fff 50%, #fff0 100%);
  visibility: visible;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: flex-start;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #fff0 0%, #fff 50%);
  visibility: visible;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: default, isNextHidden: false, isPrevHidden: true and scrollIndicatorPosition: undefined, theme: dark  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #fff;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: flex-start;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #0e1418 50%, #0e141800 100%);
  visibility: hidden;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: flex-start;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #0e141800 0%, #0e1418 50%);
  visibility: visible;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: default, isNextHidden: true, isPrevHidden: false and scrollIndicatorPosition: undefined, theme: light  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #000;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: flex-start;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #fff 50%, #fff0 100%);
  visibility: visible;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: flex-start;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #fff0 0%, #fff 50%);
  visibility: hidden;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: default, isNextHidden: true, isPrevHidden: true and scrollIndicatorPosition: undefined, theme: light  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #000;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: flex-start;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #fff 50%, #fff0 100%);
  visibility: hidden;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: flex-start;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #fff0 0%, #fff 50%);
  visibility: hidden;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: surface, isNextHidden: false, isPrevHidden: false and scrollIndicatorPosition: center, theme: dark  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #fff;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #262b2e 50%, #262b2e00 100%);
  visibility: visible;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #262b2e00 0%, #262b2e 50%);
  visibility: visible;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: surface, isNextHidden: false, isPrevHidden: false and scrollIndicatorPosition: center, theme: light  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #000;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #f2f2f2 50%, #f2f2f200 100%);
  visibility: visible;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #f2f2f200 0%, #f2f2f2 50%);
  visibility: visible;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;

exports[`getComponentCss() should return correct css for gradientColorScheme: surface, isNextHidden: false, isPrevHidden: false and scrollIndicatorPosition: center, theme: light-electric  1`] = `
":host {
  display: block;
  height: inherit !important;
}
.root {
  display: grid;
  grid-template-columns: 2em minmax(0, 1fr) 2em;
  margin: 0 -0.25rem;
  height: inherit;
}
.scroll-area {
  min-height: 1.5rem;
  grid-area: 1 / 1 / 1 / -1;
  padding: 0.25rem;
  overflow: scroll hidden;
  -ms-overflow-style: none;
  scrollbar-width: none;
}
.scroll-area::-webkit-scrollbar {
  display: none;
}
.scroll-wrapper {
  outline: transparent solid 1px;
  outline-offset: 2px;
  position: relative;
  display: inline-flex;
  min-width: 100%;
  vertical-align: top;
}
.scroll-wrapper::-moz-focus-inner {
  border: 0;
}
.scroll-wrapper:focus {
  outline-color: #000;
}
.scroll-wrapper:focus:not(:focus-visible) {
  outline-color: transparent;
}
.trigger {
  position: absolute;
  top: 0;
  bottom: 0;
  width: 1px;
  visibility: hidden;
}
.trigger:first-of-type {
  left: 0;
}
.trigger:last-of-type {
  right: 0;
}
.action-prev {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 1 / 1 / 1;
  justify-content: flex-start;
  background: linear-gradient(90deg, #f2f2f2 50%, #f2f2f200 100%);
  visibility: visible;
}
.action-prev .button::before {
  left: 0;
}
.action-next {
  position: relative;
  padding: 0.25rem 0;
  pointer-events: none;
  display: flex;
  align-items: center;
  grid-area: 1 / 3 / 1 / 3;
  justify-content: flex-end;
  background: linear-gradient(90deg, #f2f2f200 0%, #f2f2f2 50%);
  visibility: visible;
}
.action-next .button::before {
  right: 0;
}
.button {
  pointer-events: auto;
  position: static;
}
.button::before {
  content: \\"\\";
  position: absolute;
  top: 0;
  bottom: 0;
  width: max(2rem, 80%);
}"
`;
